File: node/transcribe_any_stream_audio_format.js
const fs = require("fs");
const { SpeechClient } = require("@soniox/soniox-node");

/**
 * Set your Soniox Cloud API key:
 * from command line: export SONIOX_API_KEY=<YOUR-API-KEY>
 * or
 * pass config object: new SpeechClient({ api_key: "<YOUR-API-KEY>" })
 */
const speechClient = new SpeechClient();

(async function () {
  const onDataHandler = async (result) => {
    console.log(result.words.map((word) => word.text).join(""));
  };

  const onEndHandler = (error) => {
    if (error) {
      console.log(`Transcription error: ${error}`);
    }
  };

  // transcribeStream returns object with ".writeAsync()" and ".end()" methods
  // use them to send data and end stream when done.
  const stream = speechClient.transcribeStream(
    {
      model: "en_v2_lowlatency",
      audio_format: "pcm_s16le",
      sample_rate_hertz: 16000,
      num_audio_channels: 1,
      include_nonfinal: true
    },
    onDataHandler,
    onEndHandler
  );

  // Here we simulate the stream by reading a file in small chunks.
  const CHUNK_SIZE = 1024;
  const readable = fs.createReadStream("../test_data/test_audio_long.raw", {
    highWaterMark: CHUNK_SIZE,
  });

  // Simulate data streaming
  for await (const chunk of readable) {
    await stream.writeAsync(chunk);
  }

  stream.end();
})();

File: python/transcribe_any_stream_audio_format.py
from typing import Iterable
from soniox.transcribe_live import transcribe_stream
from soniox.speech_service import SpeechClient


def iter_audio() -> Iterable[bytes]:
    with open("../test_data/test_audio_long.raw", "rb") as fh:
        while True:
            audio = fh.read(1024)
            if len(audio) == 0:
                break
            yield audio


def main():
    with SpeechClient() as client:
        for result in transcribe_stream(
            iter_audio(),
            client,
            model="en_v2_lowlatency",
            include_nonfinal=True,
            audio_format="pcm_s16le",
            sample_rate_hertz=16000,
            num_audio_channels=1,
        ):
            # Variable result contains final and non-final words.
            print("".join(w.text for w in result.words))


if __name__ == "__main__":
    main()
